### ISSUE TITLE
Development flow for 1.0 with npm link

### ISSUE THREAD

**Issue by rewop** [gatsbyjs/gatsby#731]
I am looking into the amazing implementation of 1.0. Really great job. 

I am considering to use gatsby 1.0 in one of my projects. And I would like to start contributing. However I am finding hard to `npm link` my local gatsby repository to my test repository. 

The problem is that the plugins are installed in the test repository, but required in [gatsby bootstrap](https://github.com/gatsbyjs/gatsby/blob/c9778dae0e2031c021652f3383e114287b21775f/packages/gatsby/lib/bootstrap/index.js#L127-L127) file. 

This means that I have this folder structure:
```
/Development
   /gatsby
      /packages
         /gatsby --> npm linked
            -- here gatsby-source-filesystem is not available --
         /gatsby-source-filesystem
    /test-repo
       /node_modules
          /gatsby -> ../../../../../.node/lib/node_modules/gatsby
          /gatsby-source-filesystem
```

These are the commands I am executing:
In gatsby folder,  
- `yarn install`
- `lerna bootstrap`
- `yarn run build`
- `lerna exec npm link`
- `lerna run watch --no-sort --stream --concurrency 50` (this could be updated in watch script)

In test repository folder I have both `gatsby` and `gatsby-source-filesystem` as dependecies, but since I am debugging gatsby I just link it:
- `npm link gatsby`

My test repo has loads `gatsby-source-filesystem` as plugin in `gatsby-config.js`.

I wonder, how should I set up my machine to contribute to gatsby 1.0 and debug problems while I implement my website?
I can create a PR to update the CONTRIBUTING.md file in 1.0 branch if needed :)

**Comment by noahlange** [comment]
I ended up writing a bash script to rsync the contents of each package in the `./packages` directory into the test repo's `node_modules` folder. I started with `npm link` but ran into the same issues you're encountering.

```bash
basedir="/Users/Noah/code"
repodir="${basedir}/github/gatsby-dev"
projdir="${repodir}/examples/gatsbygram"

cd "${repodir}/packages"
for dir in `ls ${repodir}/packages`;
do
  echo "copying contents of ${dir} to ${projdir}/node_modules/${dir}"
  rsync -avr --exclude="${dir}/node_modules" $dir "${projdir}/node_modules"
done
cd "$basedir"
```

Then I just run `. sync.sh` and it syncs everything over. You'll obviously want to update `$basedir`, `$repodir` and `$projdir` as appropriate to your situation.

Best of luck!

**Comment by KyleAMathews** [comment]
Someone actually just started yesterday updating the contributing guide for 1.0 :-) Please give feedback on @blargity's PR!

https://github.com/gatsbyjs/gatsby/pull/727

This is a complex problem which the naive `npm link` behavior doesn't solve. This RFC for Yarn summarizes the problem and a potential solution nicely https://github.com/yarnpkg/rfcs/blob/master/text/0000-yarn-knit.md

What I do is overly manual but at least works perfectly which is to hardlink built JavaScript files from packages to sites. Then run `npm run watch` in the root of a package to start babel watching the files in `/src`. Then create a hardlink (I've been using https://github.com/qwzybug/hln — I think symlinks cause trouble as Node.js and others will follow the symlinks right?) from the built file you're working on to its equivalent in the site you're working on e.g. `hln /path/to/checked/out/version/gatsby/packages/gatsby-source-filesystem/gatsby-node.js /my/site/node_modules/gatsby-source-filesystem/gatsby-node.js`

Once the `yarn knit` command is built, that seems like it'd be perfect for us. But until then, it might make sense to make a little hacky command line tool to setup the watcher and hardlink (or just copy) built files (using `.npmignore`) over to the site you're working on.

**Comment by rewop** [comment]
@KyleAMathews @noahlange Thanks for your feedbacks. Than this issue will be solved by #727.

I will move my feedbacks there.

**Comment by rewop** [comment]
resolved via #727
